thirdeye/thirdeye-frontend/app/pods/manage/alerts/performance/route.js

```diff
@@ -0,0 +1,228 @@
+/**
+ * Handles the 'create alert' route nested in the 'manage' route.
+ * @module self-serve/create/route
+ * @exports alert create model
+ */
+import Ember from 'ember';
+import fetch from 'fetch';
+import RSVP from 'rsvp';
+import { checkStatus, pluralizeTime, buildDateEod, parseProps, postProps } from 'thirdeye-frontend/helpers/utils';
+
+/**
+ * If true, this reduces the list of alerts per app to 2 for a quick demo.
+ */
+const demoMode = false;
+
+/**
+ * Fetches all anomaly data for found anomalies - downloads all 'pages' of data from server
+ * in order to handle sorting/filtering on the entire set locally. Start/end date are not used here.
+ * @param {Array} anomalyIds - list of all found anomaly ids
+ * @returns {Ember.RSVP promise}
+ */
+const fetchAppAnomalies = (alertList) => {
+  const alertPromises = [];
+  const startDate = buildDateEod(3, 'month').toISOString();
+  const endDate = buildDateEod(1, 'day').toISOString();
+  const tuneParams = `start=${startDate}&end=${endDate}`;
+
+  alertList.forEach((alert) => {
+    let { name, id } = alert;
+    let getAlertPerfHash = {
+      id,
+      name,
+      data: fetch(`/detection-job/eval/filter/${alert.id}?${tuneParams}`).then(checkStatus)
+    };
+    alertPromises.push(Ember.RSVP.hash(getAlertPerfHash));
+  });
+
+  return Ember.RSVP.allSettled(alertPromises);
+};
+
+/**
+ * Associate each anomaly with an application name. This is done by:
+ * 1) For each existing application, find all alert groups associated with it
+ * 2) Add the app name to each of the group's function Ids (making sure we don't duplicate an Id)
+ * @param {Array} allApps - list of all applications
+ * @param {Array} validGroups - all alert groups that are active
+ * @returns {Array} appBucket
+ */
+const fillAppBuckets = (allApps, validGroups) => {
+  let appBucket = [];
+
+  allApps.forEach((app) => {
+    let associatedGroups = validGroups.filter(group => group.application.toLowerCase().includes(app.application));
+    if (associatedGroups.length) {
+      let uniqueIds = Array.from(new Set([].concat(...associatedGroups.map(group => group.emailConfig.functionIds))));
+      if (demoMode) {
+        uniqueIds = uniqueIds.slice(0, 2);
+      }
+      if (uniqueIds.length) {
+        uniqueIds.forEach((id) => {
+          appBucket.push({ name: app.application, id });
+        });
+      }
+    }
+  });
+
+  return appBucket;
+};
+
+/**
```

Comment:
I think math functions should be extracted to a math util and imported here for modularity.

Reply:
No, this is a very route-specific grouping function - not a Math util. In this form, it would not be used anywhere else. This is where the aggregation of anomalies by application happens - core to this route.